jdbc-source/src/test/java/org/springframework/cloud/stream/module/jdbc/source/JdbcSourcePropertiesTests.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.jdbc.source;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.EnvironmentTestUtils;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Configuration;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * @author Thomas Risberg
+ */
+public class JdbcSourcePropertiesTests {
+
+	@Rule
+	public ExpectedException thrown= ExpectedException.none();
+
+	@Test
+	public void queryIsRequired() {
+		thrown.expect(BeanCreationException.class);
+		thrown.expectMessage("Field error in object 'target' on field 'query': rejected value [null]");
+		AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+		context.register(Conf.class);
+		context.refresh();
```

Comment:
`AnnotationConfigApplicationContext` is `Closeable`, so should be `close()`'d in the end, independently of test result.
At least STS barks here :smile: